### PR TITLE
Facebook support for email_in_user_when_upgrading

### PR DESCRIPTION
**Pull request by protocol86**
Adds support for Facebook when autolinking an existing user to a new federated user. Tested with Google and Facebook auth credentials for now.

### CODE REVIEW

File: FirebaseAuthUI/FUIAuth.m [hunk 1]
```diff
@@ -295,18 +295,25 @@ - (void)signInWithEmailHint:(NSString *)emailHint
...
-    if ([federatedProviderID isEqualToString:FIRGoogleAuthProviderID]) {
-      id<FUIAuthProvider> googleProviderUI;
+    if (error) {
+      completion(nil, error);
+      return;
+    }
+    NSString *existingFederatedProviderID = [self federatedAuthProviderFromProviders:providers];
+    // Set of providers which can be auto-linked
+    NSSet *supportedProviders =
+        [NSSet setWithObjects:FIRGoogleAuthProviderID,FIRFacebookAuthProviderID, nil];
```
Comment (morganchen12) [inline FirebaseAuthUI/FUIAuth.m]: nit: space after first comma

File: FirebaseAuthUI/FUIAuth.m [hunk 2]
```diff
@@ -295,18 +295,25 @@ - (void)signInWithEmailHint:(NSString *)emailHint
...
+        if ([provider.providerID isEqualToString:existingFederatedProviderID]) {
+          authProviderUI = provider;
           break;
         }
       }
-      [googleProviderUI signOut];
-      [googleProviderUI signInWithDefaultValue:emailHint
+      [authProviderUI signOut];
+      [authProviderUI signInWithDefaultValue:emailHint
                       presentingViewController:presentingViewController
```
Comment (morganchen12) [inline FirebaseAuthUI/FUIAuth.m]: nit: fix indents

File: FirebaseAuthUI/FUIAuth.m (hunk 1 above)
Comment (protocol86) [reply FirebaseAuthUI/FUIAuth.m]: Done, thanks

File: FirebaseAuthUI/FUIAuth.m (hunk 2 above)
Comment (protocol86) [reply FirebaseAuthUI/FUIAuth.m]: Done, thanks

**Pull request merged**